src/corehost/cli/apphost/bundle/header.cpp

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "bundle_runner.h"
+#include "header.h"
+#include "pal.h"
+#include "error_codes.h"
+#include "trace.h"
+#include "utils.h"
+
+using namespace bundle;
+
+bool header_t::is_valid()
+{
+    return m_data.major_version == m_current_major_version &&
+           m_data.minor_version == m_current_minor_version &&
+           m_data.num_embedded_files > 0;
+}
+
+header_t* header_t::read(FILE* stream)
+{
+    header_t* header = new header_t();
+
+    // First read the fixed size portion of the header
+    bundle_runner_t::read(&header->m_data, sizeof(header->m_data), stream);
+    if (!header->is_valid())
+    {
+        trace::error(_X("Failure processing application bundle."));
+        trace::error(_X("Bundle header version compatibility check failed"));
+
+        throw StatusCode::BundleExtractionFailure;
+    }
+
+    // bundle_id is a component of the extraction path
+    size_t bundle_id_length = 
+        bundle_runner_t::get_path_length(header->m_data.bundle_id_length_byte_1, stream);
```

Comment:
`bundle_id` is actually a part of a path (that is, it should be path-compatible, and conform to length limitation). The `get_path_length` method only reads two bytes in a length encoding (because of `MAX_PATH`) restriction. So, that's why I had called it `get_path_length` instead of `get_string_length` 

I can rewrite the function more generically, call it `read_string_length` and do the `MAX_PATH` check elsewhere -- but it doesn't serve much purpose other than the symmetry with `read_string()` function.

Reply:
OK - it was just a nit anyway... :-)